### ISSUE TITLE
proposal: cmd: add sign-list & verify-list, for all tags in the image

### ISSUE THREAD

**Issue by Dentrax** [sigstore/cosign#409]
**TL;DR:** _I propose adding two new commands to cosign, called "sign-list" and "verify-list"._

**Abstract**

As we already know that Registry HTTP API V2 exports [Listing Image Tags](https://docs.docker.com/registry/spec/api/#listing-image-tags) endpoint, calling by `GET /v2/<name>/tags/list`, we quickly list all the given tags by giving an image name. `$ crane ls <IMAGE>` for example. Signing and verifying all of the image tags would be a great feature.

**Proposal**

The most important difference from the as-is workflow is that we do not need to pass [multi-ref-args](https://github.com/sigstore/cosign/blob/8479fbbba0ac9fc2f027f43dbe67c425e9b2c34e/cmd/cosign/cli/verify.go#L137) to `verify` command anymore. Instead, we can use the new `verify-list` command to traverse and verify all the tags. It should be the same for the `sign-list` command. For example, if we want to verify all tags by using `$ cosign verify <IMAGE:tag1> <IMAGE:tag2> ... <IMAGE:tagN>`, simply passing only the image ref would be fine. If any of gets failed during signing or verifying, we should throw `exit 1`.

**Implementation**

 Since we do not support [multi-ref signing](https://github.com/sigstore/cosign/blob/8479fbbba0ac9fc2f027f43dbe67c425e9b2c34e/cmd/cosign/cli/sign.go#L184) yet, we may need to create a new function to pass the image refs:

```bash
func SignListCmd(ctx context.Context, so SignOpts, imageRefs []string) # i.e.
```

Firstly, `sign-list` should do a `[GET]` request to `/v2/<name>/tags/list` endpoint in order to _list_ all the tags. After that, inside an inner loop, it should call the `SignCmd` concurrently.

**Risks & Mitigations**

* What if an error occurs during the signing phase? Should we publish all the refs at the end of the whole signing process?
* What if there are over 100+/1000+ tags? Should we increase the concurrency level according to the tag count?
* Should we allow passing multi `<IMAGE>`s to _list_ commands?
* Will Table Writer required to print out results?

**Usage**

* sign-list:
```bash
$ cosign -key cosign.key sign-list <IMAGE>
```

* verify-list:

```bash
$ cosign -key cosign.pub verify-list <IMAGE>
```

Thanks! Waiting your feedback!
